src/bindings/c/tests/ov_infer_request_test.cpp

```diff
@@ -334,8 +334,14 @@ TEST_P(ov_infer_request_test, infer_async_wait_for) {
     OV_ASSERT_OK(ov_infer_request_start_async(infer_request));
 
     if (!HasFatalFailure()) {
-        OV_EXPECT_OK(ov_infer_request_wait_for(infer_request, 10));
-
+        ov_status_e ret = ov_status_e::OK;
+        EXPECT_NO_THROW(ret = ov_infer_request_wait_for(infer_request, 10));
+        size_t max_times = 10;
+        // Random timeout in some platform, increase wait() times if timeout occurr.
+        while (ret != ov_status_e::OK && max_times-- > 0) {
+            EXPECT_NO_THROW(ret = ov_infer_request_wait_for(infer_request, 10));
+        }
+        EXPECT_EQ(ret, ov_status_e::OK);
```

Comment:
Maybe better to increase timeout from 10 to e.g. 100? Or check both statuses `OK` and `RESULT_NOT_READY`?

Reply:
Increasing timeout to 100 is another solution, but current PR can not only get the same effect but also can check multiple-calling `wait_for()` whether has any problem if timeout occurs.

`ov_infer_request_wait_for()` has 2 possible return value: `OK` or `RESULT_NOT_READY`, we can check the return value to decide whether need more `wait_for()` iteration.